### DISCUSSION TITLE
Implementing Sponsorblock for skipping non-music segments

### DISCUSSION THREAD

**Discussion by euuurgh** [gokadzev/Musify#214] [Ideas]
I think we are all aware that for some songs, there is either a small segment in front, or after the music-video. I personally find these quite annoying, but thankfully there is a convenient way to get rid of them.

Sponsorblock, a browser add-on as its name implies to skip sponsored segments on YouTube, has a category that indexes these skippable sequences. The timestamps are crowdsourced, and therefore broadly available, especially for very popular songs.

As far as I understand Musify gets its Music from YouTube, and should therefore be compatible with Sponsorblock.

There are multiple different libraries for the Sponsorblock API, in Python, Node, Rust and Go
I am aware that this would mean adding another language to this project, but in my humble opinion it is a tradeoff worth taking.

A few things to keep in mind:

- Downloaded Music
   It would be nice to have the timestamps of downloaded music also downloaded and skipped automatically. There should also be a option to scan for new segments, if these get added by the community after the download occurs.

- Submitting timestamps
   Although possible through the API, I do not think this feature is worthwhile, you can just do it from the browser. If someone wants to implement it sure, I am just bringing this possibility up for discussion.

I am excited to see what all of you think

**Comment by gokadzev** [comment]
> I think we are all aware that for some songs, there is either a small segment in front, or after the music-video. I personally find these quite annoying, but thankfully there is a convenient way to get rid of them.
> 
> Sponsorblock, a browser add-on as its name implies to skip sponsored segments on YouTube, has a category that indexes these skippable sequences. The timestamps are crowdsourced, and therefore broadly available, especially for very popular songs.
> 
> As far as I understand Musify gets its Music from YouTube, and should therefore be compatible with Sponsorblock.
> 
> There are multiple different libraries for the Sponsorblock API, in Python, Node, Rust and Go I am aware that this would mean adding another language to this project, but in my humble opinion it is a tradeoff worth taking.
> 
> A few things to keep in mind:
> 
> * Downloaded Music
>   It would be nice to have the timestamps of downloaded music also downloaded and skipped automatically. There should also be a option to scan for new segments, if these get added by the community after the download occurs.
> * Submitting timestamps
>   Although possible through the API, I do not think this feature is worthwhile, you can just do it from the browser. If someone wants to implement it sure, I am just bringing this possibility up for discussion.
> 
> I am excited to see what all of you think

Hello, first of all thank you for the perfectly described idea! Musify already had implemented sponsorblock support and then because of other problem i removed bunch of features including sponsorblock. Since that issue is fixed now, I think it's possible to add sponsorblock support again :)


However, i can't promise about downloaded music or submitting timestamps from Musify.

**Reply by gokadzev** [reply]
Update: SponsorBlock will be available in the next release

**Reply by euuurgh** [reply]
Oh wow, thank you very much for the quick reply. I was wondering why nobody requested this feature in the past, but now I know. Thank you very much for this fast of a resolution to the issue
